### ISSUE TITLE
Error with Integer matrix and nested Float64 KroneckerProduct

### ISSUE THREAD

**Issue by vsaase** [MichielStock/Kronecker.jl#78]
Hi,
here is a minimal example:
`kronecker(Int64.(ones(4,4)), kronecker(ones(2,2), ones(2,2))) * ones(16)` which produces this error:

```
ERROR: MethodError: no method matching _kron_mul_fast_square!(::Array{Float64,1}, ::Array{Float64,1}, ::Tuple{Array{Int64,2},Array{Float64,2},Array{Float64,2}})
```

This seems to be due to mismatch in the eltypes. The error disappears if I use Float64 for the first matrix like so
`kronecker(ones(4,4), kronecker(ones(2,2), ones(2,2))) * ones(16)`

**Comment by MichielStock** [comment]
Indeed, mixed eltypes do not go together with the fast multiply code of @emerali. Fix is proposed in a #81.

**Comment by MichielStock** [comment]
fixed by merging #81, should work now!
